temporal-sdk/src/test/java/io/temporal/workflow/activityTests/LongLocalActivityWorkflowTaskHeartbeatFailureTest.java

```diff
@@ -35,46 +36,55 @@
 
   private final TestActivitiesImpl activitiesImpl = new TestActivitiesImpl();
 
+  private static final int REPLAY_COUNT = 2;
+  private static final int ACTIVITY_SLEEP_SEC = 3;
+  private static final int WORKFLOW_TASK_TIMEOUT_SEC = 2;
+
   @Rule
   public SDKTestWorkflowRule testWorkflowRule =
       SDKTestWorkflowRule.newBuilder()
           .setWorkflowTypes(TestLongLocalActivityWorkflowTaskHeartbeatFailureWorkflowImpl.class)
           .setActivityImplementations(activitiesImpl)
-          .setTestTimeoutSeconds(15)
+          .setTestTimeoutSeconds(REPLAY_COUNT * ACTIVITY_SLEEP_SEC + 10)
           .build();
 
   /**
-   * Test that local activity is not lost during replay if it was started before forced
-   * WorkflowTask.
+   * Test that local activity that failed to heartbeat and executed longer than Workflow Task
+   * Timeout will be repeated during replay
    */
   @Test
   public void testLongLocalActivityWorkflowTaskHeartbeatFailure() {
+    Preconditions.checkState(
+        ACTIVITY_SLEEP_SEC > WORKFLOW_TASK_TIMEOUT_SEC,
+        "We make sure that activity sleeps longer than the workflow task timeout to emulate heartbeat failure");
+
     WorkflowOptions options =
         WorkflowOptions.newBuilder()
             .setWorkflowRunTimeout(Duration.ofMinutes(5))
-            .setWorkflowTaskTimeout(Duration.ofSeconds(2))
+            .setWorkflowTaskTimeout(Duration.ofSeconds(WORKFLOW_TASK_TIMEOUT_SEC))
             .setTaskQueue(testWorkflowRule.getTaskQueue())
             .build();
-    TestWorkflows.TestWorkflow1 workflowStub =
+    TestWorkflows.TestWorkflowReturnString workflowStub =
         testWorkflowRule
             .getWorkflowClient()
-            .newWorkflowStub(TestWorkflows.TestWorkflow1.class, options);
-    String result = workflowStub.execute(testWorkflowRule.getTaskQueue());
+            .newWorkflowStub(TestWorkflows.TestWorkflowReturnString.class, options);
+    String result = workflowStub.execute();
+    //Shouldn't this workflow never successfully finish, because local activity suppose the fail the hearbeat every single time?
```

Comment:
@vitarb @Sushisource @mfateev I need a sanity check here. It looks like we have a local activity that doesn't heartbeat and sleeps longer than workflow task timeout. Why does such a workflow even successfully completes at all?